### PR TITLE
Wrap possible NullPointerException

### PR DESCRIPTION
**Pull request by arthurscchan**
This PR provides a possible fix for issue [#124](https://github.com/FasterXML/jackson-datatypes-collections/issues/124) by adding null checking and wrapping possible NullPointerException thrown from underlying code or upstream libraries.

### CODE REVIEW

File: guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/GuavaCollectionDeserializer.java [hunk 1]
```diff
@@ -132,15 +132,19 @@ public Object deserializeWithType(JsonParser p, DeserializationContext ctxt,
...
+            // Should usually point to START_ARRAY
+            if (p.isExpectedStartArrayToken()) {
+                return _deserializeContents(p, ctxt);
+            }
+            // But may support implicit arrays from single values?
+            if (ctxt.isEnabled(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY)) {
+                return _deserializeFromSingleValue(p, ctxt);
+            }
+            return (T) ctxt.handleUnexpectedToken(_valueClass, p);
+        } catch (NullPointerException e) {
```
Comment (cowtowncoder) [inline guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/GuavaCollectionDeserializer.java:145]: No. We'd need to know why NPE was thrown and prevent that instead -- a unit test reproduction would be useful.

**Comment by yawkat** [comment]
if you added the null check why do you need the try-catch still?

File: guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/GuavaCollectionDeserializer.java (hunk 1 above)
Comment (arthurscchan) [reply guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/GuavaCollectionDeserializer.java]: I have added in unit test and also fix the logic of how to handle those NPE in https://github.com/FasterXML/jackson-datatypes-collections/pull/125/commits/3f3f61f66a43d0964693d74cd4aac3b513cdf7a1.

File: guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/GuavaCollectionDeserializer.java [hunk 2]
```diff
@@ -132,15 +132,19 @@ public Object deserializeWithType(JsonParser p, DeserializationContext ctxt,
...
+            throw new JsonProcessingException(e);
```
[the remark this answers is not in the document]
Comment (arthurscchan) [reply guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/GuavaCollectionDeserializer.java]: Have changed the logic and add in unit test in https://github.com/FasterXML/jackson-datatypes-collections/commit/3f3f61f66a43d0964693d74cd4aac3b513cdf7a1.

**Comment by arthurscchan** [comment]
@yawkat Thanks for your comments. They are two fixes in different packages. But I have changed the logic to have null check in both cases and added in unit test.

**Review by cowtowncoder — APPROVED** [review]
Looks overall pretty good! I'll check out some details of null-handling but overall this works.

**Comment by cowtowncoder** [comment]
Thank you @arthurscchan !

**Pull request merged**